### ISSUE TITLE
Allow Pathing Exclusion

### ISSUE THREAD

**Issue by miversen33** [nvim-pack/nvim-spectre#55]
Both Ripgrep and The Silver Searcher support "ignore" patterns that can be used to exclude files/directories from the query path. Consider implementing so we can actively filter out unwanted results when trying to mass update large amounts of files.

At a quick glance, it looks like this could be done by adding a bit more interactivity 
[here for ag](https://github.com/nvim-pack/nvim-spectre/blob/4a4cf2c981b077055ef7725959d13007e366ba23/lua/spectre/search/ag.lua#L14)
[here for rg](https://github.com/nvim-pack/nvim-spectre/blob/4a4cf2c981b077055ef7725959d13007e366ba23/lua/spectre/search/rg.lua#L18)

Note:
[Silver Search Man:](https://github.com/ggreer/the_silver_searcher/blob/master/doc/ag.1.md) **--ignore PATTERN**
[Ripgrep Man:](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#manual-filtering-globs) **--glob !PATTERN**

**Comment by miversen33** [comment]
I dont know much about neovim plugin development (or lua) but this doesn't appear super difficult (at face value). I will see if I can get a PR put together sometime this week for this as a sort of proof of concept

**Comment by miversen33** [comment]
Disregard, reading is hard it seems. Thanks for the fantastic plugin!

**Comment by fredrikaverpil** [comment]
@miversen33 I'm also finding it hard to read, it seems. Did you ever figure this out?

**Comment by miversen33** [comment]
@fredrikaverpil boy this is a real denvercoder moment huh?
![image](https://github.com/nvim-pack/nvim-spectre/assets/2640668/c4ca24b6-88ec-4272-a101-3ab06ffda00e)

Anyway! I took a look at "Past Mike's" commits on my dotfiles hoping I would find something to get you an answer and I saw a whole lot of nothing.

Took a shot in the dark and attempted to just append the `ignore` arguments as part of the search query

- Without the ripgrep ignore arg
![image](https://github.com/nvim-pack/nvim-spectre/assets/2640668/6767d98c-dbf9-472e-9ff0-368382d09e3b)

- With the ripgrep ignore arg
![image](https://github.com/nvim-pack/nvim-spectre/assets/2640668/147bf4b4-3682-411a-a8ab-521d9a003087)

So basically Past Mike figured out that you just append the cli args directly to your search. And failed to put that in this thread lol.
